### PR TITLE
Use `importlib` in favor of deprecated `pkg_resources`

### PR DESCRIPTION
**Pull request by jayceslesar**
## Description

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Bad practice to be using just `import pkg_resources` instead of `import importlib.resources as pkg_resources`, as this led to my org having trouble installing and explicitly needing to add setuptools as a dependency which doesnt quite make sense. This is a drop in replacement and has existed since python 3.7. 

The issue arose from having `gradio` installed in an application my org is developing, and running it resulted in 

```
gradio/flagging.py", line 17, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

This is a drop in replacement.

Closes: #(issue)
N/A

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

**Comment by jayceslesar** [comment]
@abidlabs going to be a little more involved, I will patch the rest of `pkg_resources` stuff

**Comment by jayceslesar** [comment]
@abidlabs should be good now!

**Comment by abidlabs** [comment]
Thanks @jayceslesar! It turns out that we didn't even need that version check in `flagging.py` so I've removed it and made a few other tweaks. Will merge in once CI passes. Thanks again

**Comment by jayceslesar** [comment]
> 

Absolutely! Ill see what I can do about getting types in a good spot to get CI to pass

File: gradio/routes.py [hunk 1]
```diff
@@ -4,6 +4,11 @@
 from __future__ import annotations
 
 import asyncio
+import sys
+if sys.version_info >= (3, 9):
+    from importlib.resources import files
+else:
+    from importlib_resources import files
```
Comment (abidlabs) [inline gradio/routes.py:11]: Hmm we'll need to add `importlib_resources` as a dependency if its not available in python 3.8

Comment (jayceslesar) [reply gradio/routes.py:11]: good catch

**Comment by jayceslesar** [comment]
@abidlabs thanks for the cleanup!

**Comment by abidlabs** [comment]
No worries! I'll fix the final typing issues and then merge once CI passes

**Comment by abidlabs** [comment]
Ok I think we should be good to go. I've just asked @yvrjsharma to test on a Windows machine as well to make sure all the paths resolve correctly

**Comment by yvrjsharma** [comment]
I have just run couple of demos locally on my Windows machine, and all work fine.

**Comment by abidlabs** [comment]
Thanks @jayceslesar and @yvrjsharma will merge in after CI passes

**Pull request merged**